ha/k8s_setup/ha_setup.py

```diff
@@ -187,11 +188,14 @@ def process(self):
         Process config command.
         """
         try:
-            consul_endpoint = Conf.get(self._index, f'cortx{_DELIM}external{_DELIM}consul{_DELIM}endpoints[0]')
-            if not consul_endpoint:
+            consul_endpoints = Conf.get(self._index, f'cortx{_DELIM}external{_DELIM}consul{_DELIM}endpoints')
+            # search for supported consul endpoint url from list of configured consule endpoints
+            consul_endpoint = list(filter(lambda x: urlparse(x).scheme == const.consule_scheme, consul_endpoints))
+            if not consul_endpoint or len(consul_endpoint[0]) == 0:
                 sys.stderr.write(f'Failed to get consul config. consul_config: {consul_endpoint}. \n')
                 sys.exit(1)
-
+            # take first endpoint as it will be always 1
+            consul_endpoint = consul_endpoint[0]
```

Comment:
Where this consul endpoint are used ? Can't we pass list of consul endpoints instead one ?

Reply:
so after filter, there will be always 1 endpoint in the actual scenario.
and I check using list but somewhere it breaks and gives error that list cannot be cplit

Reply:
Resolve